server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UserRoutesTest.java

```diff
@@ -573,6 +581,38 @@ void puttingWithDomainPartInUsernameTwoTimesShouldNotBeAllowed() {
             assertThat(users).containsExactly(ImmutableMap.of("username", USERNAME_WITH_DOMAIN.asString()));
         }
 
+        @Test
+        void putShouldFailWhenConflictWithAlias(RecipientRewriteTable recipientRewriteTable) throws Exception {
+            recipientRewriteTable.addAliasMapping(MappingSource.fromUser(USERNAME_WITH_DOMAIN),
+                OTHER_USERNAME_WITH_DOMAIN.asString());
+
+            given()
+                .body("{\"password\":\"password\"}")
+            .when()
+                .put(USERNAME_WITH_DOMAIN.asString())
+            .then()
+                .statusCode(HttpStatus.CONFLICT_409)
+                .body("statusCode", is(HttpStatus.CONFLICT_409))
+                .body("type", is(ErrorResponder.ErrorType.WRONG_STATE.getType()))
+                .body("message", is("'username@domain' already have associated mappings: alias:other@domain"));
+        }
+
+        @Test
+        void putShouldFailWhenConflictWithGroup(RecipientRewriteTable recipientRewriteTable) throws Exception {
+            recipientRewriteTable.addGroupMapping(MappingSource.fromUser(USERNAME_WITH_DOMAIN),
+                OTHER_USERNAME_WITH_DOMAIN.asString());
+
+            given()
+                .body("{\"password\":\"password\"}")
+            .when()
+                .put(USERNAME_WITH_DOMAIN.asString())
+            .then()
+                .statusCode(HttpStatus.CONFLICT_409)
+                .body("statusCode", is(HttpStatus.CONFLICT_409))
+                .body("type", is(ErrorResponder.ErrorType.WRONG_STATE.getType()))
+                .body("message", is("'username@domain' already have associated mappings: group:other@domain"));
+        }
+
```

Comment:
Where is the test for:

```
GIVEN a user marketing@linagora.com
WHEN I create the user marketing@linagora.com
THEN it fails
```
?

Reply:
This test already exist prior to my changeset.